.github/workflows/terraform-observe_release.yaml

```diff
@@ -16,7 +16,20 @@ on:
         default: '0'
 
 jobs:
+  check-branch:
+    runs-on: ubuntu-latest
+    outputs:
+      branch: ${{ steps.get_branch.outputs.branch }}
+    steps:
+      - name: get_branch
+        id: get_branch
+        run: |
+          echo "branch=$(echo ${GITHUB_REF##*/})" >> $GITHUB_OUTPUT
+          echo "will only release if branch is main: ${GITHUB_REF##*/}"
```

Comment:
Seems like the logic here should be "error" and not "skip." In which case this does not need to be a separate job. A simple step in the existing job that bails if it's not the default branch will do the job.